src/lib/components/Button/Button.jsx

```diff
@@ -28,8 +29,9 @@ export const Button = React.forwardRef((props, ref) => {
     color,
     ...restProps
   } = props;
-
-  const context = useContext(ButtonGroupContext);
+  const inputGroupContext = useContext(InputGroupContext);
+  const buttonGroupContext = useContext(ButtonGroupContext);
+  const primaryContext = buttonGroupContext ?? inputGroupContext;
```

Comment:
I think we do not want to support including a button in both `inputGroup` and `buttonGroup` If that happens I would prefer to end with an error. 

I like a nice `throw` better then producing something that might differ from what the user expects.
What do others think?

Reply:
`buttonGroupContext && inputGroupContext` should be sufficient. E.g., if `Button` is used inside the `InputGroup` component, the value of `inputGroupContext` is `{layout: 'vertical', size: 'medium'}`, while `inputGroupContext?.current` is `undefined`.